### PR TITLE
Change Commit Checker to get the PR last commit

### PR DESCRIPTION
**Pull request by rimolive**
**Description of your changes:**

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: .github/workflows/commit-check-pr.yml [hunk 1]
```diff
@@ -62,15 +62,15 @@ jobs:
     needs: fetch-data
     env:
       PR_NUMBER: ${{ needs.fetch-data.outputs.pr_number }}
+      PR_HASH: ${{ needs.fetch-data.outputs.head_sha }}
     steps:
       - uses: actions/checkout@v3
       - name: Get Commits
```
Comment (DharmitD) [inline .github/workflows/commit-check-pr.yml:68]: I would suggest adding a step to include `fetch-depth: 0` here, to fetch all history for all branches and tags.

Comment (DharmitD) [reply .github/workflows/commit-check-pr.yml:68]: or a  `git fetch --all`

Comment (HumairAK) [reply .github/workflows/commit-check-pr.yml:68]: I don't think we need all branch history for this, we're just comparing main branch with the pr sha commit

**Comment by dsp-developers** [comment]
Commit Checker results:
==== These are the results of the commit checker scans         ====
==== If these are not commits from upstream kfp, then          ====
==== please ensure you adhere to the commit checker formatting ====
$(podman run -q -v /home/runner/work/data-science-pipelines/data-science-pipelines:/src/app-root quay.io/rmartine/commitchecker:latest --start 7127a61ee732921a693fccb79216b2bc74428cb5 --end 47fb475d26b820a8d3c384ef6809d9e132401d22)

**Pull request merged**